charmcraft.yaml

```diff
@@ -12,6 +12,13 @@ bases:
 
 parts:
   charm:
+    build-snaps:
+      - astral-uv
+    
+    override-build: |
+      make generate-requirements
```

Comment:
This change creates the requirements.txt on charmcraft pack or building the charm. I noticed that when I run `ARGS=--keep-models make integration` and I `juju debug-log --include mimir` then I get the error that `cannot import module crossplane` which was specified in the `pyproject.toml`.

1. When I `juju ssh mimir/0 cat /var/lib/juju/agents/unit-mimir-0/charm/requirements.txt | grep crossplane` I can see that crossplane exists
2. When I `juju ssh mimir/0 ls /var/lib/juju/agents/unit-mimir-0/charm/venv | grep crossplane` I can see that crossplane does not exist.

This means that the make command in the charmcraft.yaml works in creating the requirements.txt but its contents are not respected.

Reply:
I dont know why this would occur, but just in case have you tried doing a `charmcraft clean` before packing?  maybe there's something leftover in the cache

Reply:
Same result unfortunately, will have to investigate some more.